### PR TITLE
fix(launcher): Disable continue button on load

### PR DESCRIPTION
**Pull request by jyasveer**
- Don't populate the application name by default
- Let the user enter it
- Disable the `continue` button by default.
- Includes tests

**Addresses** https://github.com/openshiftio/openshift.io/issues/3607

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @jyasveer for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

File: src/app/space/add-app-overlay/add-app-overlay.component.ts [hunk 1]
```diff
@@ -23,14 +23,15 @@ import { Application, DeploymentApiService } from '../create/deployments/service
 })
 export class AddAppOverlayComponent implements OnDestroy, OnInit {
   currentSpace: Space;
-  isProjectNameValid: boolean;
+  isProjectNameValid = false;
```
Comment (joshuawilson) [inline src/app/space/add-app-overlay/add-app-overlay.component.ts]: why did you remove the type?

File: src/app/space/add-app-overlay/add-app-overlay.component.ts [hunk 2]
```diff
@@ -23,14 +23,15 @@ import { Application, DeploymentApiService } from '../create/deployments/service
...
   loggedInUser: User;
   projectName: string = '';
   selectedFlow: string = '';
   spaces: Space[] = [];
   subscriptions: Subscription[] = [];
   applications: string[] = [];
-  isProjectNameAvailable: boolean;
+  isProjectNameAvailable = false;
```
Comment (joshuawilson) [inline src/app/space/add-app-overlay/add-app-overlay.component.ts]: needs a type

File: src/app/space/add-app-overlay/add-app-overlay.component.ts (hunk 1 above)
Comment (jyasveer) [reply src/app/space/add-app-overlay/add-app-overlay.component.ts]: when we assign a primitive value, type is not needed by default. Type is defined from the primitive value we assign

File: src/app/space/add-app-overlay/add-app-overlay.component.ts (hunk 2 above)
Comment (jyasveer) [reply src/app/space/add-app-overlay/add-app-overlay.component.ts]: when we assign a primitive value, type is not needed by default. Type is defined from the primitive value we assign

File: src/app/space/add-app-overlay/add-app-overlay.component.ts (hunk 1 above)
Comment (joshuawilson) [reply src/app/space/add-app-overlay/add-app-overlay.component.ts]: I prefer the verbose style but yes you are right, that works too.
